sermant-plugins/sermant-flowcontrol/flowcontrol-common/src/main/java/io/sermant/flowcontrol/common/xds/retry/ExceptionRetryConditionType.java

```diff
@@ -0,0 +1,97 @@
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.flowcontrol.common.xds.retry;
+
+import io.sermant.core.utils.StringUtils;
+import io.sermant.flowcontrol.common.xds.retry.condition.ConnectErrorRetryCondition;
+import io.sermant.flowcontrol.common.xds.retry.condition.ResetBeforeRequestErrorCondition;
+import io.sermant.flowcontrol.common.xds.retry.condition.ResetErrorCondition;
+import io.sermant.flowcontrol.common.xds.retry.condition.ServerExceptionCondition;
+import io.sermant.flowcontrol.common.xds.retry.condition.SpecificHeaderNameErrorRetryCondition;
+
+import java.util.Optional;
+
+/**
+ * Retry Condition Manager
+ *
+ * @author zhp
+ * @since 2024-11-29
+ */
+public enum ExceptionRetryConditionType {
+    /**
+     * The type of conditional judgment for server errors
+     */
+    SERVER_ERROR("5xx", new ServerExceptionCondition()),
+
+    /**
+     * The type of conditional judgment for reset errors
+     */
+    RESET_ERROR("reset", new ResetErrorCondition()),
+
+    /**
+     * The type of conditional judgment for resetting errors before request
+     */
+    RESET_BEFORE_REQUEST_ERROR("reset-before-request", new ResetBeforeRequestErrorCondition()),
+
+    /**
+     * The type of conditional judgment for connect errors
+     */
+    CONNECT_ERROR("connect-failure", new ConnectErrorRetryCondition()),
+
+    /**
+     * The type of conditional judgment for Specify response headers
+     */
+    SPECIFIC_HEADER_NAME_ERROR("retriable-headers", new SpecificHeaderNameErrorRetryCondition());
+
+    /**
+     * the name of retry condition
+     */
+    private final String conditionName;
+
+    /**
+     * the instance of implements class for retry condition
+     */
+    private final RetryCondition retryCondition;
+
+    ExceptionRetryConditionType(String conditionName, RetryCondition retryCondition) {
+        this.conditionName = conditionName;
+        this.retryCondition = retryCondition;
+    }
+
+    public String getConditionName() {
+        return conditionName;
+    }
+
+    public RetryCondition getRetryCondition() {
+        return retryCondition;
+    }
+
+    /**
+     * get the instance of implements class by condition name
+     *
+     * @param conditionName condition name
+     * @return instance of implements class for retry condition
+     */
+    public static Optional<RetryCondition> getRetryConditionByName(String conditionName) {
+        for (ExceptionRetryConditionType retryConditionType : ExceptionRetryConditionType.values()) {
+            if (StringUtils.equals(retryConditionType.getConditionName(), conditionName)) {
+                return Optional.of(retryConditionType.getRetryCondition());
+            }
+        }
+        return Optional.empty();
+    }
+}
```

Comment:
Use EnumMap with key：conditionName and value：RetryCondition and initail it in statics block

Reply:
done